packages/sources/nft-blue-chip/src/transport/marketcap.ts

```diff
@@ -0,0 +1,288 @@
+      // If it wasn't in flight and coalescing is disabled, register it as in flight
+      const ttl =
+        config.MARKETCAP_TRANSPORT_MAX_RATE_LIMIT_RETRIES *
+        config.MARKETCAP_TRANSPORT_MS_BETWEEN_RATE_LIMIT_RETRIES
+      logger.debug('Setting up rest transport, setting request in flight in cache')
+      await this.inFlightCache.set(
+        this.inFlightPrefix + req.requestContext.cacheKey,
+        true,
+        ttl + 100,
+      ) // Can't use Infinity for things like Redis
+    }
+
+    logger.trace('Check if we are under rate limits to perform request')
+    const checkForRateLimit = async () => {
+      return this.waitUntilUnderRateLimit({
+        maxRetries: config.MARKETCAP_TRANSPORT_MAX_RATE_LIMIT_RETRIES,
+        msBetweenRetries: config.MARKETCAP_TRANSPORT_MS_BETWEEN_RATE_LIMIT_RETRIES,
+      })
+    }
+    await checkForRateLimit()
+
+    const provider = new ethers.providers.JsonRpcProvider(config.ETHEREUM_RPC_URL, RPC_CHAIN_ID)
+
+    logger.trace('Fetch all collection data async')
+
+    const collectionDataPromises = Promise.all(
+      NFT_COLLECTIONS.map((collection) => {
+        const tokenContract = new ethers.Contract(collection.token, BoredApeYachtClub, provider)
+        const feedContract = new ethers.Contract(collection.feed, EACAggregatorProxy, provider)
+
+        return Promise.all([
+          collection.name,
+          tokenContract.totalSupply(),
+          feedContract.latestAnswer(),
+          feedContract.decimals(),
+        ])
+      }),
+    )
+
+    const ethUsdFeed = new ethers.Contract(ETH_USD_FEED_PROXY, EACAggregatorProxy, provider)
+
+    const ethUsdDataPromises = Promise.all([ethUsdFeed.latestAnswer(), ethUsdFeed.decimals()])
+
+    logger.trace('Await data responses')
+
+    const [collectionData, ethUsdData] = await Promise.all([
+      collectionDataPromises,
+      ethUsdDataPromises,
+    ])
+
+    logger.trace('Compute total market cap')
+
+    let totalMarketcapEth = new Decimal(0)
+
+    for (const collection of collectionData) {
+      const name = collection[0]
+      const totalSupply = collection[1].toString()
+      const ethFloorPrice = collection[2].toString()
+      const decimals = collection[3].toString()
+
+      const collectionMarketcap = new Decimal(totalSupply)
+        .mul(ethFloorPrice)
+        .div(new Decimal(10).pow(decimals))
+
+      logger.info({ name, totalSupply, ethFloorPrice, decimals, collectionMarketcap })
+
+      totalMarketcapEth = totalMarketcapEth.add(collectionMarketcap)
+    }
+
+    const ethUsdPrice = ethUsdData[0].toString()
+    const ethUsdDecimals = ethUsdData[1].toString()
+
+    const totalMarketcapUsd = totalMarketcapEth
+      .mul(ethUsdPrice)
+      .div(new Decimal(10).pow(ethUsdDecimals))
+      .toNumber()
+
+    logger.info({ totalMarketcapEth, totalMarketcapUsd })
+
+    const response: AdapterResponse<MarketcapTransportGenerics['Response']> = {
+      statusCode: 200,
+      data: totalMarketcapUsd,
```

Comment:
Should this be `data: { totalMarketcapUsd }` to match the usual format of data as an object?

Reply:
Yes, I realized this should be `data: { result: totalMarketcapUsd }` instead. Updated 👍